devito/finite_differences/lazy.py

```diff
@@ -0,0 +1,49 @@
+__all__ = ['Evaluable']
+
+
+class Evaluable(object):
```

Comment:
More for future, would it be good to have a state that knows whether it is already evaluated or not?

Reply:
perhaps we should be using cached_properties ? 
not sure how would a state be useful